test/js/src/main/scala/zio/test/mock/TestEnvironment.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.mock
+
+import scala.concurrent.ExecutionContext
+
+import zio._
+import zio.clock.Clock
+import zio.console.Console
+import zio.internal.PlatformLive
+import zio.random.Random
+import zio.scheduler.Scheduler
+import zio.system.System
+
+case class TestEnvironment(
+  clock: TestClock,
+  console: TestConsole,
+  random: TestRandom,
+  scheduler: TestScheduler,
+  system: TestSystem,
+) extends Clock
+    with Console
+    with Random
+    with Scheduler
+    with System
+
+object TestEnvironment {
+
+  val Value: Managed[Nothing, TestEnvironment] =
```

Comment:
One more organizational tip:

Let's rename `TestEnvironment` => `TestEnvironmentInternals`, and make it package private (`private[test]`).

Then inside `zio.test.mock` package object, we can do:

```scala
package object mock {
  val TestEnvironment: TestEnvironment = TestEnvironmentInternals.Value
}
```

Then a user has type `TestEnvironment` and can use value `TestEnvironment`, which is more Scala-idiomatic.

Reply:
Hmm, actually, I'm not so sure about this, because `TestEnvironment.Value` has type `Managed[...]`. So let's leave this the way it is. 

We can add this value in `package object mock`:

```scala
val TestEnvironmentManaged: Managed[TestEnvironment] = TestEnvironment.Value
```

In fact, it probably makes sense to rename `TestEnvironment.Value` to `TestEnvironment.ManagedDefault` (a managed, default test environment).